sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/HDFSMetadataLog.scala

```diff
@@ -277,10 +295,34 @@ class HDFSMetadataLog[T <: AnyRef : ClassTag](sparkSession: SparkSession, path:
     for (batchId <- batchIds if batchId > thresholdBatchId) {
       val path = batchIdToPath(batchId)
       fileManager.delete(path)
+      if (metadataCacheEnabled) batchCache.remove(batchId)
       logTrace(s"Removed metadata log file: $path")
     }
   }
 
+
+  /**
+   * List the available batches on file system. As a workaround for S3 inconsistent list, it also
```

Comment:
Given that https://aws.amazon.com/blogs/aws/amazon-s3-update-strong-read-after-write-consistency/, this looks a little misleading. Could you elaborate a little more about what you are referring by `S3 inconsistent list`, please?

Reply:
I think this comment is out of date.  Amazon now delivers strong read after write consistency.  I will remove in a subsequent PR